### PR TITLE
fix(email): restrict email change. Fixes MA-173

### PR DESCRIPTION
**Pull request by WikiRik**

### CODE REVIEW

File: middlewares/members.js [hunk 1]
```diff
@@ -161,6 +161,12 @@ exports.triggerEmailChange = async (req, res) => {
         return errors.makeBadRequestError(res, 'No new email is provided.');
     }
 
+    // eslint-disable-next-line no-useless-escape
+    const EMAIL_REGEX = new RegExp('\@(' + constants.RESTRICTED_EMAILS.join('|').trim() + ')');
+    if (EMAIL_REGEX.test(req.body.new_email)) {
```
Comment (serge1peshcoff) [inline middlewares/members.js:166]: `if (constants.RESTRICTED_EMAILS.some(email => req.body.new_email.includes(email))` would work better imo

Comment (serge1peshcoff) [reply middlewares/members.js:166]: not sure if the syntax is right here, but the idea should work

Comment (WikiRik) [reply middlewares/members.js]: Thanks for the suggestion! I've changed it

**Comment by serge1peshcoff** [comment]
what do you think of moving this to MailChange's validation? this seems more right, as IMO these kind of checks do not belong in controllers, but in models instead

**Comment by WikiRik** [comment]
> what do you think of moving this to MailChange's validation? this seems more right, as IMO these kind of checks do not belong in controllers, but in models instead

That's actually a very good idea. I forgot that MailChange has its own model.

**Pull request merged**